app/views/templates/_form.html.erb

```diff
@@ -43,16 +43,25 @@
               <input type="radio" name="options" id="option1" ><%= _("Code") %>
             </label>
             <label class="btn btn-default btn-sm" onclick="set_preview()">
-              <input type="radio" name="options" id="option2" ><%= _("Preview") %>
+              <input type="radio" name="options" id="option2" ><%= _("Diff") %>
+            </label>
+            <label class="btn btn-default btn-sm" onclick="set_render()">
+              <input type="radio" name="options" id="option3" ><%= _("Preview") %>
             </label>
           </div>
+          <span id="preview_host_selector" style="display:none">
+            <%= select_tag :preview_host_id, options_from_collection_for_select(Host.authorized(:view_hosts).limit(100), :id, :name),
```

Comment:
Good points! Order makes sense, managed hosts make sense for Ptable and ProvisioningTemplate, leaving without it for other potential templates. Associated OSes does not seem to work nicely here, note that until we have the template saved, we don't have the associations so we couldn't display any host - usually the case for new templates or if there are no host of such OS yet. Snippets don't even have associations (but e.g. networking one is OS specific). Also previewing templates might be useful for host with other OS just to see if it works.